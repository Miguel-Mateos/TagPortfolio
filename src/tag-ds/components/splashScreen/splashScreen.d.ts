import React from 'react';
export interface ISplashScreen {
    /**
    * Set a fixed colour as a background for splash screen component
    */
    backgroundColor?: string;
    /**
    * Set an image as a background for splash screen component
    */
    backgroundImageURI?: string;
    /**
    * Set an video/animation as a background for splash screen component
    */
    backgroundVideoURI?: string;
    /**
    * Set an image of a person or company
    */
    logoURI: string;
    /**
    * Add class for splash screen component
    */
    className?: string;
    [others: string]: any;
}
export declare const SplashScreen: React.FC<ISplashScreen>;
