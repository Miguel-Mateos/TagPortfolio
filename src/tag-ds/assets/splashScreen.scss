.tag-ds {
    @media (max-width: $mobile) {
        .splash-screen, &.splash-screen {
            width: 100vw;
            height: 100vh;
            overflow: hidden;
        }
        .logo-wrapper {
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%)
        }
        .logo-image {
            width: calc(8 * var(--unit));
            height: calc(8 * var(--unit));
        }
        .video-wrapper video {
            width: 100vw;
            height: 100%;
        }
    }
}