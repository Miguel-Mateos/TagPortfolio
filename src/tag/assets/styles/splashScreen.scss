.tag-ds {
    @media (max-width: $mobile) {
        .splash-screen, &.splash-screen {
            width: 100vw;
            height: 100vh;
            overflow: hidden;
        }
        .logo-wrapper {
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%)
        }
        .logo-image {
            width: calc(8 * var(--unit));
            height: calc(8 * var(--unit));
        }
        .video-wrapper video {
            width: 100vw;
            height: 100%;
        }
        .splash-screen-logo {
            animation: fade-in 1s ease-in-out;
        }
    }
}

@keyframes fadeIn {
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}
