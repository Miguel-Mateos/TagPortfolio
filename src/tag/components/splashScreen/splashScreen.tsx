import React, { useEffect, useState } from 'react'

export interface ISplashScreen {
  /**
   * Set a fixed colour as a background for splash screen component
   */
  backgroundColor?: string
  /**
   * Set an image as a background for splash screen component
   */
  backgroundImageURI?: string
  /**
   * Set an video/animation as a background for splash screen component
   */
  backgroundVideoURI?: string
  /**
   * Set an image of a person or company
   */
  logoURI: string
  /**
   * Add class for splash screen component
   */
  className?: string
  [others: string]: any
}

export const SplashScreen: React.FC<ISplashScreen> = (props) => {
  const {
    backgroundColor,
    backgroundImageURI,
    backgroundVideoURI,
    logoURI,
    className,
    children,
    ...rest
  } = props

  const [mobile, setMobile] = useState<boolean>(window.innerWidth < 672)

  useEffect(() => {
    window.addEventListener('resize', () => {
      if (window.innerWidth < 672) setMobile(true)
      else setMobile(false)
    })
    return () => window.removeEventListener('resize', () => null)
  }, [])

  const logo = (
    <div
      className="logo-wrapper"
      data-testid={rest ? `${rest['data-testid']}-logo-wrapper` : undefined}
    >
      <img
        src={logoURI}
        alt="logo"
        className="logo-image"
        data-testid={rest ? `${rest['data-testid']}-logo-image` : undefined}
      />
    </div>
  )

  const backgroundVideo = (
    <div
      data-testid={rest ? `${rest['data-testid']}-video-wrapper` : undefined}
      className="video-wrapper"
    >
      <video autoPlay muted loop>
        <source src={backgroundVideoURI} type="video/mp4" />
      </video>
    </div>
  )

  if (mobile) {
    return (
      <div
        className="tag-ds splash-screen"
        data-testid={rest ? `${rest['data-testid']}-container` : undefined}
        style={
          backgroundColor
            ? { backgroundColor }
            : backgroundImageURI
            ? { backgroundImage: `url(${backgroundImageURI})` }
            : undefined
        }
      >
        {backgroundVideoURI && backgroundVideo}
        {logo && logo}
      </div>
    )
  } else {
    return <></>
  }
}
