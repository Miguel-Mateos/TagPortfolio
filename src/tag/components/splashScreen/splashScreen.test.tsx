import React from 'react';
import {ISplashScreen, SplashScreen} from './splashScreen';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';

const imageURI = 'https://www.w3schools.com/howto/img_avatar.png';
const backgroundImageURI = 'https://www.w3schools.com/html/img_girl.jpg';
const backgroundVideoURI =
  'https://assets.mixkit.co/videos/preview/mixkit-clouds-in-the-sky-flowing-with-the-wind-21584-large.mp4';

const SplashScreenTest = (props: ISplashScreen) => {
  window.innerWidth = 400;
  return <SplashScreen logoURI={props.logoURI} data-testid="splash-screen" className="tag-ds" />;
};

test('splash screen component should render', () => {
  const component = renderer.create(<SplashScreenTest backgroundImageURI={backgroundImageURI} logoURI={imageURI} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('render default splash screen', async () => {
  const {getByTestId} = render(<SplashScreenTest logoURI={imageURI} />);
  const component = getByTestId('splash-screen-container');
  expect(component).toBeDefined();
});

test('render splash screen with background image', async () => {
  const {getByTestId} = render(<SplashScreenTest logoURI={imageURI} backgroundImageURI={backgroundImageURI} />);
  const component = getByTestId('splash-screen-container');
  expect(component).toBeDefined();
  expect(backgroundImageURI).toBeDefined();
});

test('render splash screen with background video', async () => {
  const {container} = render(<SplashScreenTest logoURI={imageURI} backgroundVideoURI={backgroundVideoURI} />);
  const component = container.getElementsByClassName('video-wrapper');
  expect(component).toBeDefined();
});
