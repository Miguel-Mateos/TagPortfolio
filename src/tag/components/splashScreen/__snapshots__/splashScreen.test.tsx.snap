// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splash screen component should render 1`] = `
<div
  className="tag-ds splash-screen"
  data-testid="splash-screen-container"
>
  <div
    className="logo-wrapper"
    data-testid="splash-screen-logo-wrapper"
  >
    <img
      alt="logo"
      className="logo-image"
      data-testid="splash-screen-logo-image"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
  </div>
</div>
`;
